diff --git a/libc-bottom-half/headers/public/__struct_sockaddr_un.h b/libc-bottom-half/headers/public/__struct_sockaddr_un.h
index 6371194..511954b 100644
--- a/libc-bottom-half/headers/public/__struct_sockaddr_un.h
+++ b/libc-bottom-half/headers/public/__struct_sockaddr_un.h
@@ -5,6 +5,7 @@
 
 struct sockaddr_un {
     __attribute__((aligned(__BIGGEST_ALIGNMENT__))) sa_family_t sun_family;
+    char sun_path[30];
 };
 
 #endif
