diff --git a/libc-top-half/musl/include/arpa/inet.h b/libc-top-half/musl/include/arpa/inet.h
index f657c30..5b4b526 100644
--- a/libc-top-half/musl/include/arpa/inet.h
+++ b/libc-top-half/musl/include/arpa/inet.h
@@ -13,20 +13,15 @@ uint16_t htons(uint16_t);
 uint32_t ntohl(uint32_t);
 uint16_t ntohs(uint16_t);
 
-#ifdef __wasilibc_unmodified_upstream /* WASI has no inet_addr */
 in_addr_t inet_addr (const char *);
-in_addr_t inet_network (const char *);
 char *inet_ntoa (struct in_addr);
-#endif
 int inet_pton (int, const char *__restrict, void *__restrict);
 const char *inet_ntop (int, const void *__restrict, char *__restrict, socklen_t);
 
 int inet_aton (const char *, struct in_addr *);
-#ifdef __wasilibc_unmodified_upstream /* WASI has no inet_makeaddr */
 struct in_addr inet_makeaddr(in_addr_t, in_addr_t);
 in_addr_t inet_lnaof(struct in_addr);
 in_addr_t inet_netof(struct in_addr);
-#endif
 
 #ifdef __cplusplus
 }
