diff --git a/libc-top-half/musl/include/stdio.h b/libc-top-half/musl/include/stdio.h
index d63d739..ae044aa 100644
--- a/libc-top-half/musl/include/stdio.h
+++ b/libc-top-half/musl/include/stdio.h
@@ -159,10 +159,8 @@ FILE *tmpfile(void) __attribute__((__deprecated__("tmpfile is not defined on WAS
 FILE *fmemopen(void *__restrict, size_t, const char *__restrict);
 FILE *open_memstream(char **, size_t *);
 FILE *fdopen(int, const char *);
-#ifdef __wasilibc_unmodified_upstream /* WASI has no popen */
 FILE *popen(const char *, const char *);
 int pclose(FILE *);
-#endif
 int fileno(FILE *);
 int fseeko(FILE *, off_t, int);
 off_t ftello(FILE *);
