diff --git a/libc-top-half/musl/include/stdio.h b/libc-top-half/musl/include/stdio.h
index d63d739..ae044aa 100644
--- a/libc-top-half/musl/include/stdio.h
+++ b/libc-top-half/musl/include/stdio.h
@@ -159,10 +159,8 @@ FILE *tmpfile(void) __attribute__((__deprecated__("tmpfile is not defined on WAS
 FILE *fmemopen(void *__restrict, size_t, const char *__restrict);
 FILE *open_memstream(char **, size_t *);
 FILE *fdopen(int, const char *);
-#ifdef __wasilibc_unmodified_upstream /* WASI has no popen */
 FILE *popen(const char *, const char *);
 int pclose(FILE *);
-#endif
 int fileno(FILE *);
 int fseeko(FILE *, off_t, int);
 off_t ftello(FILE *);
diff --git a/libc-top-half/musl/src/internal/stdio_impl.h b/libc-top-half/musl/src/internal/stdio_impl.h
index 7f19dd9..c547686 100644
--- a/libc-top-half/musl/src/internal/stdio_impl.h
+++ b/libc-top-half/musl/src/internal/stdio_impl.h
@@ -43,9 +43,7 @@ struct _IO_FILE {
 	size_t buf_size;
 	FILE *prev, *next;
 	int fd;
-#ifdef __wasilibc_unmodified_upstream // WASI has no popen
 	int pipe_pid;
-#endif
 #if defined(__wasilibc_unmodified_upstream) || defined(_REENTRANT)
 	long lockcount;
 #endif
